test/batch/test_crud_batch.py

```diff
@@ -852,19 +948,70 @@ def test_create_data(self):
             batch._create_data('objects', ObjectsBatchRequest())
         check_startswith_error_message(self, error, read_timeout_error_message('objects'))
         mock_connection.post.assert_called_with(
             path="/batch/objects",
             weaviate_object={'fields' : ['ALL'], 'objects': []},
         )
-        self.assertEqual(mock_connection.post.call_count, 2)
+        self.assertEqual(mock_connection.post.call_count, 3 + 1)
+
+        ## test ConnectionError, connection_error_retries = 0
+        mock_connection = mock_connection_method('post', side_effect = RequestsConnectionError('Test!'))
+        mock_connection.timeout_config = (2, 100)
+        batch = Batch(mock_connection)
+        batch.connection_error_retries = 0
+        with self.assertRaises(RequestsConnectionError) as error:
+            batch._create_data('references', ReferenceBatchRequest())
+        check_startswith_error_message(self, error, requests_error_message)
+        mock_connection.post.assert_called_with(
+            path="/batch/references",
+            weaviate_object=[],
+        )
+        self.assertEqual(mock_connection.post.call_count, 1)
+
+        ## test ConnectionError, connection_error_retries = 3 (default)
+        mock_connection = mock_connection_method('post', side_effect = RequestsConnectionError('Test!'))
+        mock_connection.timeout_config = (2, 100)
+        batch = Batch(mock_connection)
+        with self.assertRaises(RequestsConnectionError) as error:
+            batch._create_data('objects', ObjectsBatchRequest())
+        check_startswith_error_message(self, error, requests_error_message)
+        mock_connection.post.assert_called_with(
+            path="/batch/objects",
+            weaviate_object={'fields': ['ALL'], 'objects': []},
+        )
+        self.assertEqual(mock_connection.post.call_count, 3 + 1)
+
+        ## test alternating errors
+        i_for_alt_errors = 0
+        def alternating_errors():
+            nonlocal i_for_alt_errors
+            if i_for_alt_errors % 2 == 0:
+                err = RequestsConnectionError
+            else:
+                err = ReadTimeout
+            i_for_alt_errors += 1
+            raise err
+
+        mock_connection = mock_connection_method('post', side_effect = lambda path, weaviate_object: alternating_errors())
+        mock_connection.timeout_config = (2, 100)
+        batch = Batch(mock_connection)
+        batch.connection_error_retries = 2
+        batch.timeout_retries = 2
+        with self.assertRaises(RequestsConnectionError) as error:
+            batch._create_data('objects', ObjectsBatchRequest())
+        check_startswith_error_message(self, error, requests_error_message)
+        mock_connection.post.assert_called_with(
+            path="/batch/objects",
+            weaviate_object={'fields': ['ALL'], 'objects': []},
+        )
+        self.assertEqual(mock_connection.post.call_count, 2 + 2 + 1)
```

Comment:
This also looks like something parametrization can help with

Reply:
Lets leave it as it is for now too.